src/main/java/org/apache/maven/plugins/javadoc/AbstractJavadocMojo.java

```diff
@@ -3799,7 +3799,7 @@ else if ( SystemUtils.IS_OS_MAC_OSX && !JavaVersion.JAVA_SPECIFICATION_VERSION.i
         {
             javadocExe = new File( SystemUtils.getJavaHome() + File.separator + "bin", javadocCommand );
         }
-        // On Java 9, this has moved to: /usr/lib/jvm/java-9-openjdk-amd64/bin/javadoc
+        // On Java 9 the jre subdirectory was removed from the JDK

```

Comment:
I am not a maintainer, but I think this comment could still go a bit more into details.

"On Java 9 the jre subdirectory was removed from the JDK" does not explain why the relative path from javaHome is different. The interesting part (as explained in the issue) is:
If the env_var `JAVA_HOME` was not set, and the java binary was called from a JDK installation from the jre subfolder, "java.home" was previously set to `JDK_HOME/jre`. Now as the jre directory disappeared, it is more obvious why this change is necessary.

Maybe a maintainer can tell what should go here.

Reply:
It was intended to be an addendum to the comment about 15 lines above above, which I think includes the details your're looking for:

```
        // Try to find javadocExe from System.getProperty( "java.home" )
        // By default, System.getProperty( "java.home" ) = JRE_HOME and JRE_HOME
        // should be in the JDK_HOME
```

I can add a longer comment if you think it is worthwhile.
```

Reply:
@wardev please add a longer comment as requested by @bmhm 
I feel it is clearer for future changes, when Java < 9 will be history